doc/components.md

```diff
@@ -7,45 +7,62 @@ This diagram might look intimidating, but don't worry, it is not as complex as i
+The `onMount` method can be overridden to run initialization code for the component. When this
+method is called, BaseGame ensures that all the mixins which would change this component's behavior
+are already resolved.
 
-The `onRemove` method can be overridden to run code before the component is removed from the game, it is only run once even if the component is removed both by using the `BaseGame` remove method and the ´Component´ remove method.
+The `onRemove` method can be overridden to run code before the component is removed from the game,
+it is only run once even if the component is removed both by using the `BaseGame` remove method and
+the ´Component´ remove method.
 
-The `onLoad` method can be overridden to run asynchronous initialization code for the component, like loading an image for example. This method is executed after the initial "preparation" of the component is run, meaning that this method is executed after `onMount` and just before the inclusion of the component on the `BaseGame` list of components.
+The `onLoad` method can be overridden to run asynchronous initialization code for the component,
+like loading an image for example. This method is executed after the initial "preparation" of the
+component is run, meaning that this method is executed after `onMount` and just before the inclusion
+of the component in the `BaseGame`'s list of components.
 
 ## BaseComponent
-Usually if you are going to make your own component you want to extend `PositionComponent`, but if you want to be able to handle effects and child components but handle the positioning differently you can extend the `BaseComponent`.
+Usually if you are going to make your own component you want to extend `PositionComponent`, but if
+you want to be able to handle effects and child components but handle the positioning differently
+you can extend the `BaseComponent`.
 
-It is used by `SpriteBodyComponent` and `BodyComponent` in Forge2D since those components doesn't have their position in relation to the screen, but in relation to the Forge2D world.
+It is used by `SpriteBodyComponent`, `PositionBodyComponent`, and `BodyComponent` in Forge2D since
+those components doesn't have their position in relation to the screen, but in relation to the
+Forge2D world.
 
 ### Composability of components
 
-Sometimes it is useful to make your component wrap other components. For example by grouping visual components through a hierarchy.
-You can do this by having child components on any component that extends `BaseComponent`, for example `PositionComponent` or `BodyComponent`.
-When you have child components on a component every time the parent is updated and rendered, all the children are rendered and updated with the same conditions.
+Sometimes it is useful to wrap other components inside of your component. For example by grouping
+visual components through a hierarchy. You can do this by having child components on any component
```

Comment:
Shouldn't it be Children components?

Reply:
Nope, the plural comes on the "components". :)